### PR TITLE
feat: Upstream `strongdm` component

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Upstream `strongdm` component

## why
* strongDM is a proxy that manages and audits access to databases, servers, clusters, and web apps. The strongDM network is comprised of a local client, gateway intermediary, and configuration layer.

## references
* https://www.strongdm.com/how-it-works

## notes
* This was converted from `helmfile` to terraform's `helm_release` resource
* To pass pre-commit hook

```shell
tfenv install 0.13.5
make readme
pre-commit run --show-diff-on-failure --color=always --all-files
git add modules/strongdm
```

But unfortunately it's not passing the YAML checks.

### CODE REVIEW

File: modules/strongdm/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Component: `strong-dm`
```
Comment (aknysh) [inline modules/strongdm/README.md:1]:
```suggestion
# Component: `strongdm`
```

File: modules/strongdm/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+This component installs the `strong-dm` component.
```
Comment (aknysh) [inline modules/strongdm/README.md:3]:
```suggestion
This component provisions [strongDM](https://www.strongdm.com/) gateway, relay and roles 
```

File: modules/strongdm/helm.tf [hunk 3]
```diff
@@ -0,0 +1,88 @@
+resource "helm_release" "gateway" {
+  count = local.enabled ? 2 : 0
```
Comment (aknysh) [inline modules/strongdm/helm.tf:2]: why 2? maybe make it a variable?

File: modules/strongdm/helm.tf [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  set_sensitive {
+    name  = "relayToken"
+    value = sdm_node.gateway[count.index].gateway[count.index].token
+  }
+}
+
+resource "helm_release" "relay" {
+  count = local.enabled ? 2 : 0
```
Comment (aknysh) [inline modules/strongdm/helm.tf:24]: why 2? 
do we always provision 2 of these?

File: modules/strongdm/main.tf [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+data "aws_ssm_parameter" "ssh_admin_token" {
+  count    = local.enabled ? 1 : 0
+  name     = "/vpn/sdm/ssh-admin-token"
+  provider = aws.api_keys
+}
+
+# Create a gateway
+resource "sdm_node" "gateway" {
+  count = local.install_gateway ? 2 : 0
```
Comment (aknysh) [inline modules/strongdm/main.tf:30]: we'll always have 2?

File: modules/strongdm/roles.tf [hunk 6]
```diff
@@ -0,0 +1,32 @@
+locals {
+  roles = toset(["db-user", "ssh-user", "kubernetes-user"])
+}
+
+resource "sdm_role" "all_access" {
+  count     = local.create_roles ? 1 : 0
+  name      = "all-access"
```
Comment (aknysh) [inline modules/strongdm/roles.tf:7]: `all-access` - is this name required by strongDM?
or should we use the `label` module to create unique names?

File: modules/strongdm/variables.tf [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  type        = bool
+  default     = true
+  description = "Set `true` to register nodes as SSH targets"
+}
+
+## helm-release
+
+variable "kubernetes_namespace" {
+  type        = string
+  description = "The namespace to install the release into. Defaults to `default`."
```
Comment (aknysh) [inline modules/strongdm/variables.tf:50]:
```suggestion
  description = "The Kubernetes namespace to install the release into. Defaults to `default`."
```

File: modules/strongdm/variables.tf [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  default     = null
+}
+
+variable "dns_zone" {
+  type        = string
+  description = ""
```
Comment (aknysh) [inline modules/strongdm/variables.tf:56]: add `description`

**Review by aknysh — COMMENTED** [review]
please see comments

File: modules/strongdm/helm.tf (hunk 3 above)
Comment (nitrocode) [reply modules/strongdm/helm.tf:2]: The original gateway had 2 instances so I just reused the count logic.

Yes, that seems like a good idea to be able to change the number of gateways. I'll add that.

File: modules/strongdm/roles.tf (hunk 6 above)
Comment (nitrocode) [reply modules/strongdm/roles.tf:7]: I'm unsure if this is required but worth looking into. A cursory google search does not seem like it's a required name.

**Comment by dylanbannon** [comment]
@nitrocode Can you indicate the earliest version of Terraform for which we need to ensure compatibility by adding one of the `terraform/[version]` tags to this PR? (Assigning a `terraform/[version]` tag should lead to pre-commit checks rerunning, pre-commit autocommitting any changes and, consequently, the pre-commit checks passing.)

**Pull request merged**